plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/NoPermissions.tsx

```diff
@@ -0,0 +1,130 @@
+import { Navigation } from '../../components/navigation/navigation';
+import { PluginCard } from './components/plugin-card/plugin-card';
+import { PluginErrorBanner } from './components/plugin-error-banner/PluginErrorBanner';
+
+/** Page to be shown when the user does not have permissions to install plugins */
+export const NoPermissionsError = ( {
+	context,
+	navigationProgress,
+	sendEvent,
+}: {
+	context: Pick<
+		CoreProfilerStateMachineContext,
+		'pluginsAvailable' | 'currentUser'
+	>;
+	sendEvent: (
+		payload:
+			| PluginsInstallationRequestedEvent
+			| PluginsPageSkippedEvent
+			| PluginsLearnMoreLinkClickedEvent
+	) => void;
+	navigationProgress: number;
+} ) => {
+	const skipPluginsPage = () => {
+		return sendEvent( {
+			type: 'PLUGINS_PAGE_SKIPPED',
+		} );
+	};
+
+	const pluginsCardRowCount = Math.ceil(
+		context.pluginsAvailable.length / 2
+	);
+
+	return (
+		<div
+			className="woocommerce-profiler-plugins"
+			data-testid="core-profiler-plugins"
+		>
+			<Navigation
+				percentage={ navigationProgress }
+				onSkip={ skipPluginsPage }
+			/>
+			<div className="woocommerce-profiler-page__content woocommerce-profiler-plugins__content">
+				<Heading
+					className="woocommerce-profiler__stepper-heading"
+					title={ __(
+						'Get a boost with our free features',
+						'woocommerce'
+					) }
+					subTitle={ __(
+						'Enhance your store by installing these free business features. No commitment required – you can remove them at any time.',
+						'woocommerce'
+					) }
+				/>
+
+				<PluginErrorBanner
+					pluginsInstallationPermissionsFailure={ true }
+				/>
+
+				<div
+					className={ clsx(
+						'woocommerce-profiler-plugins__list',
+						`rows-${ pluginsCardRowCount }`
+					) }
+				>
+					{ context.pluginsAvailable.map( ( plugin ) => {
+						const {
+							key: pluginSlug,
+							learn_more_link: learnMoreLink,
+						} = plugin;
+						return (
+							<PluginCard
+								key={ pluginSlug }
+								plugin={ plugin }
+								checked={ false }
+								disabled={ true }
+							>
+								{ learnMoreLink && (
+									<PluginCard.LearnMoreLink
+										onClick={ () => {
+											sendEvent( {
+												type: 'PLUGINS_LEARN_MORE_LINK_CLICKED',
+												payload: {
+													plugin: pluginSlug,
+													learnMoreLink,
+												},
+											} );
+										} }
+									/>
+								) }
+							</PluginCard>
+						);
+					} ) }
+				</div>
+				<div
+					className={ clsx(
+						'woocommerce-profiler-plugins__footer',
+						`rows-${ pluginsCardRowCount }`
+					) }
+				>
+					<div className="woocommerce-profiler-plugins-continue-button-container">
+						<Button
+							className="woocommerce-profiler-plugins-continue-button"
+							variant="primary"
+							onClick={ skipPluginsPage }
+						>
+							{ __( 'Continue', 'woocommerce' ) }
+						</Button>
+					</div>
+				</div>
+			</div>
+		</div>
+	);
+};
```

Comment:
Would it be better to add classes/overlay to the existing plugins screen? Just curious the tradeoff because this would be somewhat duplicated code no?

Reply:
I initially started out with branching in the existing Plugins page but ended up doing this instead because 
1. A view towards a possible direction of the error page being completely different to the plugins page
2. The visuals and interactions are both different so those would both have to branch and it's already a fair bit going on in there. Surmountable for sure, e.g just overlaying CSS for visuals and pointer-events:none but that would limit 1

So roughly the calculation I made is that the pages are more likely to diverge than to converge, and hence it'll be less constraining to split it out right now while I'm working on it rather than at a later date if/when its asked of us

Reply:
(that's a really good question and something I struggled with a bit)